pixeltable/exprs/type_cast.py

```diff
@@ -0,0 +1,53 @@
+import json
+from typing import Optional, Dict, List, Tuple, Any
+
+import sqlalchemy as sql
+
+import pixeltable.type_system as ts
+from .expr import DataRow, Expr
+from .row_builder import RowBuilder
+
+
+class TypeCast(Expr):
+    """
+    An `Expr` that represents a type conversion from an underlying `Expr` to
+    a specified `ColumnType`.
+    """
+    def __init__(self, underlying: Expr, new_type: ts.ColumnType):
+        super().__init__(new_type)
+        self.components: List[Expr] = [underlying]
+        self.id: Optional[int] = self._create_id()
```

Comment:
Side note: let's discuss whether we want to require type hints in cases where the type can be inferred.

Reply:
I think they should be optional (can be used when it helps make the code clearer or more readable, but not required).